inference-engine/tests/functional/shared_test_classes/src/single_layer/deformable_convolution.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/single_layer/deformable_convolution.hpp"
+
+namespace LayerTestsDefinitions {
+
+std::string DeformableConvolutionLayerTest::getTestCaseName(testing::TestParamInfo<deformableConvLayerTestParamsSet> obj) {
+    deformableConvSpecificParams convParams;
+    InferenceEngine::Precision netPrecision;
+    InferenceEngine::Precision inPrc, outPrc;
+    InferenceEngine::Layout inLayout, outLayout;
+    InferenceEngine::SizeVector inputShapes;
+    std::string targetDevice;
+    std::tie(convParams, netPrecision, inPrc, outPrc, inLayout, outLayout, inputShapes, targetDevice) =
+        obj.param;
+    ngraph::op::PadType padType;
+    InferenceEngine::SizeVector deformable_vals, kernel, stride, dilation;
+    std::vector<ptrdiff_t> padBegin, padEnd;
+    size_t groups, deformable_groups, convOutChannels;
+    std::tie(deformable_vals, kernel, stride, padBegin, padEnd, dilation, groups, deformable_groups, convOutChannels, padType) = convParams;
+
+    std::ostringstream result;
+    result << "IS=" << CommonTestUtils::vec2str(inputShapes) << "_";
+    result << "DV" << CommonTestUtils::vec2str(deformable_vals) << "_";
+    result << "K" << CommonTestUtils::vec2str(kernel) << "_";
+    result << "S" << CommonTestUtils::vec2str(stride) << "_";
+    result << "PB" << CommonTestUtils::vec2str(padBegin) << "_";
+    result << "PE" << CommonTestUtils::vec2str(padEnd) << "_";
+    result << "D=" << CommonTestUtils::vec2str(dilation) << "_";
+    result << "G=" << groups << "_";
+    result << "DG=" << deformable_groups << "_";
+    result << "O=" << convOutChannels << "_";
+    result << "AP=" << padType << "_";
+    result << "netPRC=" << netPrecision.name() << "_";
+    result << "inPRC=" << inPrc.name() << "_";
+    result << "outPRC=" << outPrc.name() << "_";
+    result << "inL=" << inLayout << "_";
+    result << "outL=" << outLayout << "_";
+    result << "trgDev=" << targetDevice;
+    return result.str();
+}
+
+InferenceEngine::Blob::Ptr DeformableConvolutionLayerTest::GenerateInput(const InferenceEngine::InputInfo &info) const {
+        InferenceEngine::Blob::Ptr blobPtr;
+        const std::string name = info.name();
+        if (name == "a_data") {
+            blobPtr = LayerTestsUtils::LayerTestsCommon::GenerateInput(info);
+        } else if (name == "b_defor_vals") {
+            blobPtr = FuncTestUtils::createAndFillBlob(info.getTensorDesc(), 0, 0, 0, 7235346); // TODO
```

Comment:
TODO: we need to extend SLT to test also DeformableConvolution with offsets different from 0. We intend to add this at later stage of implementation - this comment is not to forget about it.

Reply:
Done in [here](https://github.com/openvinotoolkit/openvino/pull/4312/commits/a33a9c92b2483e3ac9d79770a83d25876e4ce51al).